src/main/java/com/epam/ms/handler/BaseEventHandler.java

```diff
@@ -0,0 +1,16 @@
+package com.epam.ms.handler;
+
+import com.epam.ms.repository.domain.Event;
+import com.epam.ms.util.EmailSender;
+import com.epam.ms.util.UserServiceMediator;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public abstract class BaseEventHandler {
+    @Autowired
+    protected EmailSender emailSender;
+    @Autowired
+    protected UserServiceMediator mediator;
```

Comment:
I'd put these properties directly to implementation classes and convert `BaseEventHandler` into interface

Reply:
Could you please explain, why? Just for me to know ;)

Reply:
the question is about composition over inheritance
and you don't have any reusable code in BaseHandler. Regarding to these two properties they are singletons and can be injected in all the components as they are.
There is no need in Base parent class at all 😄

Reply:
Thank you!